imports/ui/unit-explorer/unit-explorer.jsx

```diff
@@ -58,12 +75,26 @@ class UnitExplorer extends Component {
     }
   }
 
+  get filteredUnits () {
+    const { selectedStatusFilter, sortBy, selectedRoleFilter } = this.state
+    const { unitList, currentUserId } = this.props
+    const statusFilter = selectedStatusFilter === 'All' ? unitItem => true
```

Comment:
Please break this down with a switch statement like you did in the other two places